plugins/device/rest.go

```diff
@@ -167,6 +167,39 @@ func deviceEditHandler(w http.ResponseWriter, r *http.Request) {
 	restutils.SendHTTPResponse(ctx, w, http.StatusOK, nil)
 }
 
+func deviceDeleteHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+	peerID := mux.Vars(r)["peerid"]
+	if uuid.Parse(peerID) == nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "invalid peer-id passed in url")
+		return
+	}
+	deviceName := r.URL.Query().Get("device")
+	if deviceName == "" {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "Device name not provided in URL")
+		return
+	}
+
+	txn, err := transaction.NewTxnWithLocks(ctx, peerID)
+	if err != nil {
+		status, err := restutils.ErrToStatusCode(err)
+		restutils.SendHTTPError(ctx, w, status, err)
+		return
+	}
+	defer txn.Done()
+
+	err = deviceutils.DeleteDevice(peerID, deviceName)
```

Comment:
question: don't we need to lock device before deleting it?

Reply:
#998, as @aravindavk said we need to lock device for smart volume creation, the same way we should consider locking device before deleting it.